implementations/micrometer-registry-signalfx/src/test/java/io/micrometer/signalfx/SignalFxNamingConventionTest.java

```diff
@@ -31,8 +31,11 @@ class SignalFxNamingConventionTest {
 
     @Test
     void tagKey() {
-        assertThat(convention.tagKey("_boo")).isEqualTo("boo");
-        assertThat(convention.tagKey("sf_boo")).isEqualTo("boo");
+        assertThat(convention.tagKey("_boo")).isEqualTo("a_boo");
+        assertThat(convention.tagKey("__boo")).isEqualTo("a__boo");
+        assertThat(convention.tagKey("sf_boo")).isEqualTo("asf_boo");
+        assertThat(convention.tagKey("sf__boo")).isEqualTo("asf__boo");
+        assertThat(convention.tagKey("sf_sf_boo")).isEqualTo("asf_sf_boo");
```

Comment:
I added this test:
`assertThat(convention.tagKey("sf.boo")).isEqualTo("sf_boo");`
So it seems possible to inadvertently create `sf_` prefixed dimensions.